core/src/main/kotlin/net/corda/core/serialization/amqp/ObjectSerializer.kt

```
@@ -71,11 +71,14 @@ class ObjectSerializer(val clazz: Type, factory: SerializerFactory) : AMQPSerial
         return interfaces.map { nameForType(it) }
     }
 
-
     fun construct(properties: List<Any?>): Any {
         if (javaConstructor == null) {
             throw NotSerializableException("Attempt to deserialize an interface: $clazz. Serialized form is invalid.")
         }
-        return javaConstructor.newInstance(*properties.toTypedArray())
+
+        /* remember that the properties list contain all of the serialised properties of a class, not just
```

At the moment there are no specific use cases, it came out of a discussion into what to do with classes that contained properties not set by the constructor, i.e. weather to reject them (as per classes whose constructor parameters don't all map to properties) or silently accept them (which is current behaviour although that's by coincidence rather than explicit intent) 

Adding the non constructed properties into the blob seemed like a good idea for the carpenter as it would allow it to have access to computed constant values which wouldn't be available to the carpented version.  

For example, a state might define some constant property that is just initialised in the constructor that is useful. Currently, the carpenter won't see that constant. 

It feels like this isn't the would be desirable behaviour, it might be worth limiting it to properties that only have getters (and not setters) or are constant